### PR TITLE
S2U-32 & S2U-28 Assignments and Messages: Tag support - MASTER

### PR DESCRIPTION
**Pull request by stetsche**
This incorporates the port to master for both S2U-32 and S2U-28, because they have inter-dependencies. This commit includes work of the following PRs: #11944 #11907 #11756 #11752 #11741 #11740 #11627 #11685 #11697 #11620 #11567 #11593 #11616 #12218 #12215 #12238

Conversion scripts: https://github.com/sakaiproject/sakai-reference/pull/232

https://sakaiproject.atlassian.net/browse/S2U-32
https://sakaiproject.atlassian.net/browse/S2U-28

### CODE REVIEW

**Comment by bgarciaentornos** [comment]
The vue components part comes originally from this other PR -> https://github.com/sakaiproject/sakai/pull/10484

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 1]
```diff
@@ -5530,6 +5530,9 @@
 # The email address that will receive the confirmation mails from the import jobs (by default uses the defined in portal.error.email)
 #tags.import_job_email=youremail@yourdomain.ext
 
+# S2U-32 and S2U-28 Allow the use of tags in some tools: assignments, private messages (default is true)
```
[the remark this answers is not in the document]
Comment (stetsche) [reply config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:5533]: I think question pool tags just depend on `samigo.author.usetags`

**Comment by stetsche** [comment]
@bgarciaentornos Thanks for the review. I will make a separate commit for those fixes and create a PR for s2u-22x, then the branches will diverge less.

File: vuecomponents/tool/pom.xml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>compile vue components to webcomponents bundl</id>
+                        <goals>
+                            <goal>npm</goal>
+                        </goals>
+                        <configuration>
+                            <arguments>
+				                run build
```
[the remark this answers is not in the document]
Comment (stetsche) [reply vuecomponents/tool/pom.xml:113]: I think this is fixed with S2U-35

File: vuecomponents/tool/pom.xml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                    </execution>
+                    <execution>
+                        <id>Test vue components</id>
+                        <phase>${sakai.skip.test.phase}</phase>
+                        <goals>
+                            <goal>npm</goal>
+                        </goals>
+                        <configuration>
+                            <arguments>
+				                run test
```
[the remark this answers is not in the document]
Comment (stetsche) [reply vuecomponents/tool/pom.xml:125]: I think this is fixed with S2U-35

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java [hunk 4]
```diff
@@ -3120,15 +3162,19 @@ private PrivateMessage processPvtMsgReplyAllSendHelper(boolean preview, Boolean
 	            rrepMsg.setScheduledDate(openDate);
 	            schedulerMessage(rrepMsg, isSendEmail());
 	            resetComposeContents();
+      return rrepMsg;
```
[the remark this answers is not in the document]
Comment (stetsche) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java:3165]: This return should actually not be part of this PR!

File: msgcntr/messageforums-app/src/webapp/js/messages.js [hunk 5]
```diff
@@ -92,6 +92,13 @@ function addTagSelector(obj) {
   	}
 }
 
+function initTagSelector(view) {
+    const tagsInputId = view + ":tag_selector";
```
[the remark this answers is not in the document]
Comment (stetsche) [reply msgcntr/messageforums-app/src/webapp/js/messages.js:96]: done, but not shown on this lines

File: msgcntr/messageforums-app/src/webapp/jsp/compose.jsp [hunk 6]
```diff
@@ -370,6 +374,20 @@
 						--%>
 				</h:dataTable>
 
+      <h:panelGroup rendered="#{PrivateMessagesTool.canUseTags}">
+        <h4><h:outputText value="#{msgs.pvt_tags_header}" /></h4>
```
[the remark this answers is not in the document]
Comment (stetsche) [reply msgcntr/messageforums-app/src/webapp/jsp/compose.jsp:378]: false alarm

**Review by bgarciaentornos — APPROVED** [review]
Thanks, ready to go!

**Comment by mpellicer** [comment]
This is a tremendous work, thank you very much everybody involved @stetsche @bgarciaentornos @st-manu etc

**Pull request merged**